### PR TITLE
Fix bug when max scores were negative, and feedback given

### PR DESCRIPTION
**Pull request by lm-44**
Negative max scores caused a bug

As a side-effect of the fix, error messages may not be as specific, specifically, entering a negative score for the grade command

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -65,4 +65,23 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
+    public static boolean isUnsignedInteger(String s) {
+        requireNonNull(s);
+
+        try {
+            int value = Integer.parseInt(s);
+            // "-0" and "+0" is successfully parsed by Integer#parseInt(String)
+            return value >= 0 && !s.startsWith("+") && !s.startsWith("-");
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
```
Comment (avrilmohh) [inline src/main/java/seedu/address/commons/util/StringUtil.java:85]: Good job on ensuring the in put is a valid unsigned integer and not a signed one.

Comment (avrilmohh) [inline src/main/java/seedu/address/commons/util/StringUtil.java:85]: Good job on ensuring the input is a valid unsigned integer and not a signed one.

**Review by avrilmohh — COMMENTED** [review]
Overall looks good!

File: src/main/java/seedu/address/logic/parser/AddAssignmentCommandParser.java [hunk 2]
```diff
@@ -30,24 +30,24 @@ public AddAssignmentCommand parse(String args) throws ParseException {
...
+                PREFIX_ASSIGNMENT_MAX_SCORE);
         try {
             studentIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_STUDENT_INDEX).get());
             assignmentName = ParserUtil.parseAssignmentName(argMultimap.getValue(PREFIX_ASSIGNMENT_NAME).get());
             maxScore = ParserUtil.parseMaxScore(argMultimap.getValue(PREFIX_ASSIGNMENT_MAX_SCORE).get());
         } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                    AddAssignmentCommand.MESSAGE_USAGE), pe);
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddAssignmentCommand.MESSAGE_USAGE), pe);
```
Comment (jian7490) [inline src/main/java/seedu/address/logic/parser/AddAssignmentCommandParser.java:45]: Good formatting to follow coding standard

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -118,13 +119,10 @@ public static AssignmentName parseAssignmentName(String assignmentName) throws P
...
-        int parsedScore;
-        try {
-            parsedScore = Integer.parseInt(trimmedMaxScore);
-        } catch (NumberFormatException e) {
-            throw new ParseException("The score must be an integer!");
+        if (!StringUtil.isUnsignedInteger(trimmedMaxScore)) {
+            throw new ParseException(Assignment.MAX_SCORE_MESSAGE_CONSTRAINTS);
         }
-        return parsedScore;
+        return Integer.parseInt(trimmedMaxScore);
```
Comment (jian7490) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:125]: This is a similar issue happening in gradeAssignmentCommand. Great attempt to simplify the code.

**Review by jian7490 — COMMENTED** [review]
LGTM!

**Pull request merged**